src/appmixer/utils/controls/OnError/OnError.js

```diff
@@ -1,59 +1,94 @@
-
-    const result = await context.callAppmixer({
-        endPoint: '/logs',
-        method: 'GET',
-        qs: {
-            from,
-            size,
-            sort: 'gridTimestamp:asc',
-            query: `@timestamp:[${context.state.since} TO ${now}] AND severity:error`,
-            flowId: context.flowId
-        }
-    });
+async function fetchErrorsPage(context, lock, limit, logId = null, lastGridTimestamp = null, size = 100, errors = []) {
+
+    const qs = {
+        size: Math.min(limit, size),
+        query: 'severity:error',
+        flowId: context.flowId
+    };
+
+    if (logId && lastGridTimestamp) {
+        qs.searchAfter = [logId, lastGridTimestamp];
+    }
+
+    while (true) {
+        try {
+            const result = await context.callAppmixer({
+                endPoint: '/logs',
+                method: 'GET',
+                qs
+            });
+
+            if (!(result.hits && result.hits.length > 0 && limit > result.hits.length)) {
+                break;
+            }
 
-    errors = errors.concat(result.hits);
+            lock.extend(parseInt(context.config.lockTTL, 10) || 1000 * 60 * 2);
+            const lastLog = result.hits[result.hits.length - 1];
+            errors.push(...result.hits);
 
-    if (result.hits.length === 0 || result.hits.length < size) {
-        return errors;
+            if (errors.length >= limit) {
+                break;
+            }
+
+            qs.searchAfter = [lastLog['_id'], lastLog['gridTimestamp']];
+        } catch (error) {
+            console.error('Error fetching errors:', error);
+            await context.log({ level: 'error', error: error.message || error });
+            break;
+        }
     }
 
-    return getErrors(context, now, errors, result.nextFrom, size);
+    return errors;
 }
 
-/**
- * Add Flow name and component labels to those errors.
- * @param {Context} context
- * @param {Array} errors
- */
+// Adds labels to errors for better identification
 function addLabels(context, errors) {
 
-    return errors.map(err => {
-        return Object.assign(
-            err,
-            {
-                flowName: context.flowInfo.flowName
-            },
-            err.componentId ? {
-                componentLabel: context.flowDescriptor[err.componentId].label || err.componentType.split('.')[3]
-            } : {});
-    });
+    return errors.map(err => ({
+        ...err,
+        flowName: context.flowInfo.flowName,
+        componentLabel: err.componentId ? (context.flowDescriptor[err.componentId]?.label || err.componentType.split('.')[3]) : undefined
+    }));
 }
 
 module.exports = {
 
-    async start(context) {
+    async tick(context) {
+
+        const lockName = `errorProcessingLock-${context.componentId}`; // Define a unique lock name
+        let lock;
+        try {
+            lock = await context.lock(lockName, { ttl: context.config.lockTTL || 1000 * 60 * 5, maxRetryCount: 0 });
```

Comment:
I don't see the lock being extended. If getting all logs takes more then 5 minutes (which could happen with that `while(true)` and `Infinite` limit. Then that could take more than 5 minutes. And if a second process gets here, after the TTL expires, you will get duplicates on the output port.